grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingQueryParamMatchRule.java

```diff
@@ -42,5 +42,16 @@ public enum HttpJsonTranscodingQueryParamMatchRule {
     /**
      * Uses the original fields in .proto files to match {@link QueryParams} of an {@link HttpRequest}.
      */
-    ORIGINAL_FIELD
+    ORIGINAL_FIELD,
+
+    /**
+     * Uses the original field and json name.
+     * <a href="https://protobuf.dev/programming-guides/proto3/#json">json mapping</a>
+     */
+    ORIGINAL_FIELD_JSON_NAME,
```

Comment:
`json_name` is a customized value so I think it needs to override both `ORIGINAL_FIELD` and `LOWER_CAMEL_CASE` rules. What do you think of automatically overriding a JSON field with `json_name` if it is set?

Reply:
I agree with you.
I'll try to fix it like that.
If json_name is present in the descriptor option, I can set the originField to json_name, right?

Reply:
Correct. 
- If no `json_name` is set, the conversion will follow `HttpJsonTranscodingQueryParamMatchRule`.
- If `json_name` is set, `HttpJsonTranscodingQueryParamMatchRule` is ignored and use the value as is.

Reply:
I modified it. Please check it again.
And I also changed the title.